### PR TITLE
Added parse function to retrieve the port from an address

### PR DESCRIPTION
**Pull request by AntoineRondelet**
see #404

### CODE REVIEW

File: utils/parse/parse.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package parse
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// Port returns the port from an address
+func Port(address string) (uint, error) {
```
Comment (htdvisser) [inline utils/parse/parse.go]: This doesn't support the `user:pass@host:port` scheme, which we might want to use in the future. Let's be future-proof ;)

Comment (AntoineRondelet) [inline utils/parse/parse.go]: right

File: utils/parse/parse.go [hunk 2]
```diff
@@ -8,16 +8,17 @@ import (
 
 // Port returns the port from an address
 func Port(address string) (uint, error) {
-	parts := strings.SplitN(address, ":", 2)
-	if len(parts) != 2 {
+	parts := strings.Split(address, ":")
```
Comment (johanstokking) [inline utils/parse/parse.go]: We have a builtin `url.Parse()` for this. This gives a `Host` field which you can safely check on presence of a port (format is always `host:port`)

Comment (htdvisser) [inline utils/parse/parse.go]: That only works when you supply a `scheme://`, which is not the case here.

Comment (htdvisser) [inline utils/parse/parse.go]: From https://golang.org/pkg/net/url/#URL:

> A URL represents a parsed URL (technically, a URI reference). The general form represented is:
> 
> ```
> scheme://[userinfo@]host/path[?query][#fragment]
> ```
>
> URLs that do not start with a slash after the scheme are interpreted as:
> 
> ```
> scheme:opaque[?query][#fragment]
> ```

That would be incorrect.

**Pull request merged**